pkg/base1/cockpit.js

```diff
@@ -2894,19 +2911,38 @@ function full_scope(cockpit, $, po) {
         }
 
         self.fetch = function fetch(beg, end) {
-            var opts = $.extend({ }, self.options, {
-                timestamp: timestamp(beg, interval),
-                limit: end - beg
-            });
-            if (opts.source == "direct")
-                opts.source = "pcp-archive";
-            transfer(opts, drain);
+            /* Provide some fake data synchronously to the series so
+             * that it doesn't ask us again for the same range while
+             * we are still fetching it asynchronously.
+             *
+             * XXX - is this something the series should do itself?
+             */
+            self.series.input(beg, new Array(end-beg), { });
```

Comment:
This code should be in the series. Agree with the comment. I think simply this bit of logic in the caller to fetch.